src/firebase/firebase.ts

```diff
@@ -150,6 +151,7 @@ export function* initializeCloudMessaging(app: ReactNativeFirebase.Module, addre
     yield call([app.messaging(), 'hasPermission'])
   Logger.info(TAG, 'Current messaging authorization status', authStatus.toString())
   if (authStatus === firebase.messaging.AuthorizationStatus.NOT_DETERMINED) {
+    yield take(HomeActions.VISIT_HOME)
```

Comment:
if you reviewed an earlier version of this PR you may recall that I was doing some in-memory caching on home visits in the prev version. This is no longer necessary, since this particular block will rarely get entered more than once. If it does get evaluated more than once, it means the user quit the app and came back before accepting or rejecting notifications-- and in this case, an in-memory cache wouldn't do us any good.

Reply:
This only takes a single action, right? If this function throws after this action has been taken, the [caller](https://github.com/valora-inc/wallet/blob/d61919c182998847e8fe4c02bb614a6a53cc45d6/src/firebase/saga.ts#L63) will try calling this function again in a loop, at which point we'll block forever on this `take`, since the action will not have been dispatched again, right?